src/bootstrap/builder.rs

```diff
@@ -246,6 +297,8 @@ enum ReallyDefault<'a> {
 
 pub struct ShouldRun<'a> {
     pub builder: &'a Builder<'a>,
+    module: &'a str,
```

Comment:
Can you `s/module/kind` and potentially make the type be Kind rather than eagerly converting into `&str`? TaskPath would then have `Option<Kind>` as well, and immediately catch typos in the prefix.

Reply:
Done!